client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala

```diff
@@ -913,6 +928,54 @@ class LifecycleManager(val appUniqueId: String, val conf: CelebornConf) extends
     context.reply(pbReportShuffleFetchFailureResponse)
   }
 
+  private def handleReportBarrierStageAttemptFailure(
+      context: RpcCallContext,
+      appShuffleId: Int,
+      appShuffleIdentifier: String): Unit = {
+
+    val shuffleIds = shuffleIdMapping.get(appShuffleId)
+    if (shuffleIds == null) {
+      throw new UnsupportedOperationException(s"unexpected! unknown appShuffleId $appShuffleId")
+    }
+    var ret = true
+    shuffleIds.synchronized {
+      shuffleIds.get(appShuffleIdentifier) match {
+        case Some((shuffleId, true)) =>
+          ret = invokeAppShuffleTrackerCallback(appShuffleId)
```

Comment:
IMO, the `handleReportBarrierStageAttemptFailure` function serves two purposes: 
1. unregisterMapOutput
2. unregisterShuffle

However, DAGScheduler in spark will unregister all mapOutput before resubmitting the barrier stage.  Can we unregister the previous shuffle when generating a new shuffleId for the barrier stage and remove `handleReportBarrierStageAttemptFailure`?

Reply:
You are right, now that we are only supporting this when fetch failure is `throwsFetchFailure` enabled, we dont need it : it was necessary in the earlier iteration when `throwsFetchFailure` could be `false` as well.